test/integration/harness/lib/docker_ctl.py

```
@@ -0,0 +1,54 @@
+def recreate(service: str) -> None:
+    """Force-recreate a service container. Drops the network namespace,
+    which releases any listening sockets the previous container had bound
+    — needed after a kill since `docker start` of the same container can
+    inherit a stale port binding from the killed sshd, causing the new
+    sshd's tcpip-forward to fail with "Address in use" (Bug #2 in the
+    2026-04-28 baseline report)."""
+    _compose("up", "-d", "--force-recreate", "--no-deps", service)
+
+
+def is_running(service: str) -> bool:
+    cp = subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)